File: classes/parte-4-control-del-programa/058-guardas-y-validacion-temprana/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	edad, _ := strconv.Atoi(strings.TrimSpace(line))
	if edad < 0 {
		fmt.Println("invalido")
		return
	}
	if edad < 18 {
		fmt.Println("menor")
		return
	}
	fmt.Println("adulto")
}

File: classes/parte-4-control-del-programa/058-guardas-y-validacion-temprana/implementaciones/c/main.c
#include <stdio.h>

int main(void) {
    long edad;
    if (scanf("%ld", &edad) != 1) return 1;
    if (edad < 0) {
        printf("invalido\n");
    } else if (edad < 18) {
        printf("menor\n");
    } else {
        printf("adulto\n");
    }
    return 0;
}
